docs/design/mp_pod_to_vpc_subnet.md

```diff
@@ -0,0 +1,126 @@
+<!--
+SPDX-License-Identifier: MIT
+Copyright (c) 2021 The Authors.
+
+Authors: Phu Tran          <@phudtran>
+
+-->
+
+# Mizar User defined VPCs and Subnets for Vanilla Kubernetes
+
+## Introduction
+
+This document defines the interface for users to define their own VPCs and subnets. Furthermore, it also describes the interface for assigning pods to these specific subnets and VPCs.
+
+## Custom VPC Creation
+
+When creating a VPC, users must define the VPC's name, IP, prefix, and the number of dividers.
+
+An example is given below.
+
+```yaml
+apiVersion: "mizar.com/v1"
+kind: Vpc
+metadata:
+  name: vpc0
+spec:
+  ip: "12.0.0.0"
+  prefix: "8"
+  dividers: 2
+```
+
+#### Requirements
+- The CIDR range is valid.
+- The VPC must not already exist.
+
+## Custom Subnet Creation
+
+When creating a subnet users must define the subnet's name, IP, prefix  the number of bouncers, and the VPC that it belongs to.
```

Comment:
Can we consider supporting default values for number of dividers and bouncers? (Default 1, if not specified). Effectively, it would make these fields optional.

Reply:
Sure I can add that.